### DISCUSSION TITLE
Asking for help creating custom logic

### DISCUSSION THREAD

**Discussion by bbakerman** [raphw/byte-buddy#1342] [Q&A]
I have a property getter requirement which today uses reflection.  

We start with a property name (say `foo` or `bar`) and we need to get the property from a source object

I am looking to use ByteBuddy to help generate a class to do this from a common generated class.  I am guessing it looks like this

```
class GeneratedXXX {
   static Object loadValue(String propertyName, SpecificPojoClass sourceObject) {
        if ("foo".equals(propertyName)) {
           return sourceObject.getFoo();
       else if ("bar".equals(propertyName)) {
           return sourceObject.getBar();
       else {
           return null;
       }
 }
}
```

In ByteBuddy how would I insert such custom logic.  Is this ASM generation or is there helpers that would allow me to get started here.

Is there some examples I should look at that are this kind of generated template

**Comment by raphw** [comment]
You can likely use `MethodCall` to generate such an invocation dispatch. On the other side: `MethodHandle`s do not have real overhead for these purposes. Normally, there is no need to replace such modern "reflection" with code generation as the VM handles handles quite well.
